#+PROPERTY: COOKIE_DATA todo recursive

* Letter from eLife [2017-07-22 Sat]
From: editorial@elifesciences.org<mailto:editorial@elifesciences.org>
Date: July 22, 2017 at 10:45:16 AM EDT
To: spencejr@umich.edu<mailto:spencejr@umich.edu>
Subject: eLife decision: 30-05-2017-RA-eLife-29132
Reply-To: editorial@elifesciences.org<mailto:editorial@elifesciences.org>

Dear Dr. Spence,

Thank you for submitting your article "Bacterial colonization stimulates a complex physiological response in the immature human intestinal epithelium" for consideration by eLife. Your article has been reviewed by 3 peer reviewers, and the evaluation has been overseen by a Reviewing Editor and Wendy Garrett as the Senior Editor. The following individuals involved in review of your submission have agreed to reveal their identity: Emma Slack (Reviewer #1).


The reviewers have discussed the reviews with one another and the Reviewing Editor has drafted this decision to help you prepare a revised submission. We hope you will be able to submit the revised version within two months. This time line is strict and of special importance at eLife. Also, there was consensus among the reviewers that your manuscript would be most appropriate, if adequately revised, for our Tools and Resources section.

SUMMARY:
Human intestinal organoids (HIOs) were used to investigate host microbial interactions at the epithelial surface. Using live or dead Escherichia coli they report contact and hypoxia driven responses with antimicrobial peptide production, maturation of the mucus layer and improved paracellular barrier function. The paper represents a technical development of the HIO system, with findings that are aligned with interpretations from previous culture systems.

ESSENTIAL REVISIONS:
General comments on presentation in the paper
Especially for a paper where the main point is development of an experimental system, the technical validation and reporting of the methods and results in the paper must be completely solid. The reviewers noted missing information on the number of biological and technical repeats, concentrations of stimuli, number of injected bacteria from the figure legends. The methods section was considered incomplete to allow a researcher from a related field to exactly reproduce every experiment shown - concentrations, solvents and timings - e.g. Fig 7, how much TNF, how much IFNgamma? It was unclear exactly how the false discovery rate was assessed - which algorithm was used? What sample numbers were used in the calculations and how reproducible were the experiments? Specific information about the E. coli strain should be given, and whether it is motile.

Major comments
The model depends on selective injection of live bacteria into HIOs and evidence is provided in Figure 1 about this. The experimental protocol needs to be far more explicit in the main body of the text: in the methods under 'HIO culture' it is stated 'HIOs were maintained in ENR media without antibiotics prior to microinjection experiments'; then under 'Microinjection', '..... cultures were rinsed with PBS and treated with ENR media containing penicillin and streptomycin...'. Controls that assess the potential influence of carry-over of antibiotics of the readouts reported in Figure 1 are required.

Further issues on Figure 1.
A. Fig 1B appears to be compromised by fixation and/or freezing artifacts.
B. In Figure 1C the y axis is labelled 24h delta CFU. Given that the scale becomes fractional, it seems likely that ratios rather than differences are being used.
C. What dose(s) of injections were used for 1D?
D. As a main panel of Figure 1E, showing two agar plates does not add much, especially when the legend does not give details of exactly when are how they were generated. Further, the final sentence of the third paragraph of page 4 and the results quoted in the following paragraph after day 3 are contradictory. Controls are absolutely necessary to ensure that these findings are not simply due to antibiotic carry-over.

The transcriptomic approach does not capture all basic information about the system. Does the E. coli injection alter the size, morphology and longevity of the HIOs? Does it alter the rate of epithelial cell turnover (which could be easily tested by BrdU labeling, and would be predicted from e.g. Proc Natl Acad Sci U S A. 2011 Mar 15;108 Suppl 1:4570-7)? Does it alter the pattern of epithelial cell maturation, i.e. the fraction of mature paneth cells, goblet cells, enteroendocrine cells etc?

It is accepted that enteroids are 3 dimensional structures, and the study has been productive in the evaluation of stem cell dynamics and other tissue level events. However, organoids are not vascularized with a capillary network that generate O2 gradients in vivo, thus organism level physiological events such as hypoxia are difficult to address and these caveats must be included in the discussion

Using the pathway analysis from the GO and REACTOME databases limits the insight that the reader is given into the changes in gene expression. For example, in Figure 2C 'muscle cell differentiation' is shown as a pathway. It would be possible to report detailed network analyses at least in the supporting material so that the contribution of different transcripts to the overall analysis is clear. This would allow considerable refinement of the time-dependent transcriptional response descriptions in the text.

In Figure 3C it is difficult to appreciate whether the cadherin stain has worked in the PBS panel. How reproducible are these results? What happens with a hypoxic control?

The data analysis shown in Fig 4. was complex and challenging for the reviewers to appreciate in its current form. The subset organisation must be clear and specific.
A. Why is there no PBS control with the NFkB inhibitor? Many cell survival/cell death pathways converge on NFkB signalling and the inhibitor may exert effects already at baseline which would then fall into gene set 1 and gene set 2? From the Venn-diagrams shown in panel A, it appears that genes in set 1 and 2 are mutually exclusive, which would make the lacking control less problematic, but apparently the hypoxia-responsive genes are highly enriched in both (of course these may be different genes in the same pathway? But this would be slightly surprising if a statistical exclusion had been made).
B. For the discussion of the analysis, it appears that an assumption is made that the effect of live E. coli is identical to dead E. coli combined with hypoxia. There is a large body of literature on "vital PAMPs" (e.g. Nature. 2011 May 22;474(7351):385-9), which might suggest there is more to it. Nevertheless, there appears to be an excellent correlation between genes regulated by live and dead E. coli in the plots in Fig. 4B. If the hypothesis is correct, then injecting dead E. coli and then immediately transferring the organoids to a hypoxic chamber should produce a gene-expression profile that correlates better that the dead E. coli alone. Noticeably the gradient of the correlation between hypoxia/PBS and live E. coli/PBS seems to be close to zero?
C. Is there some sort of statistical significance cut-off for the genes identified for each set in panel B? The clouds appear to very closely approach a log2-FC of zero, suggesting genes showing very small changes in expression are included? Would it be logical to show the data pre-filtered for p-value? Or show the limits of the region and color-code for p-value?
D. In panel C, is it right that "% genes matched to pathway" is this the percentage of genes from each set (e.g. the full 1940 genes in Set 1) that map to the indicated pathway? Thus 5%, i.e. around 400 genes from set 1 map to "regulation of cytoskeleton organization"? Please clarify. Also, as the plots in B suggest that many genes are included with a very small up- or down-regulation, it would be important to have some handle on not just the significance, but also the average absolute size of the change observed. A second set of graphs, or a supplementary figure with more information would be helpful.
E. In the legend, were pathways with enrichment P-values greater than 0.01 excluded?

In Fig 5B and C, it would be important to include a group microinjected with dead E. coli, immediately followed by hypoxia, to conclude that both factors act together to induce b-defensins. For Fig 5D, does hBD-2 require a reducing agent for activity as hBD1 does (Nature 469, 419-423)? Typically, these pore-forming AMPs to exert a stronger effect on the rapidly growing bacteria than in the stationary phase, and in fact, your maximum growth rate (i.e. maximum curve gradient) is even higher where the BD-2 is added, suggesting that something in the BD-2 may even permit faster E coli growth in LB. Death over several hours in late stationary phase may be rather due to accumulation of a toxic metabolite. To control for these effects, it will be important to show growth data with heat-inactivation of the BD-2. To focus on killing in the stationary phase a late stationary-phase culture could be treated with differing concentrations of BD-2 over short time-courses (including the inactivated controls), measuring loss of membrane integrity by Sytox-green uptake by flow cytometry or microscopy. As O-antigens can inhibit AMP function, E. coli K-12 could be included in these experiments.

In relation to figure 6, can induction of mucus production and induction of goblet cell differentiation be delineated? The slow appearance of mucin gene upregulation appears more consistent with a differentiation phenotype than simple gene expression?

The interpretation that 'Epithelial barrier integrity is enhanced following bacterial association' (p13) is rather at odds with the data in Figure 7 B and D where the PBS and E.coli treated permeability is the same. Is the meaning that NFkappaB signaling is required for the compensatory effects of the barrier in E.coli-treated organoids? What is the effect of the inhibitor alone?

MINOR POINTS:

The data shown in Fig 2C is potentially misleading. What you are most likely seeing here is a sigmoidal growth to a reasonably uniform carrying capacity of E. coli. The logarithmic growth rate is likely identical for all different inocula, but the plateau phase is reached earlier with the higher dose injections. Please simply plot the injected CFU and the final CFU per organoid. Ideally a time-course over the first 24h of colonization with a low and high dose of E. coli could be shown, giving you a more detailed handle on the type of growth occurring within the organoids.

What is the explanation for the increase in BD1 over time in the PBS injected organoids (Figure 2E which lacks units)? It would be interesting to include control panels here for cell death (LDH release or similar) and total cell number per sample volume.

Fucosylation of intestinal glycans has been reported to depend upon the presence of bacterial pathobionts or pathogens (e.g. SFB, Salmonella) and the activity of innate lymphoid cells in the lamina propria (e.g. Science 12 Sep 2014:Vol. 345, Issue 6202, 1254009), or to be associated with mature M cells. "Mature carbohydrate modifications" may be something of an oversimplification, and this observation may even suggest that the closed nature of the organoids leads to a non-pathogenic E. coli strain being recognized as quite aggressive.

Page 3, second paragraph. The description of the principal component analysis should read that PC1 correlates with developmental stage, while PC2 correlates with tissue maturation status. Currently the text implies that there is a direct relationship.

The authors should be accurate in acknowledging older literature (the hypoxia NFkappaB effect has been known since 1994 and the BD-2 effect since 1997) - albeit not in this context.

We will look forward to hearing from you with a revised article with tracked changes, and a response letter (uploaded as an editable file) describing the changes made in response to the decision and review comments. If source manuscript files have not already been provided, we will need them at the revision stage: http://submit.elifesciences.org/html/elife_author_instructions.html#Revised_submissions

If your work involved the use of cell lines, please indicate in the Materials and methods section of your manuscript if their identity has been authenticated, state the authentication method (such as STR profiling), and report the mycoplasma contamination testing status. Please consult our Journal Policies for further details: https://submit.elifesciences.org/html/elife_author_instructions.html#policies

If your revised article is accepted for publication, please note that authors can choose to have their accepted PDF published by eLife within a few days of acceptance, in advance of the version of record that will be subject to technical processing, editing, and author proofing.

Please consider whether this is suitable for your work, so you can select the appropriate option at the time of resubmission. In the event that the study has potentially broad public relevance, we recommend you consult your press officer before making this decision.

Please also be aware that we will publish the most substantive parts of the decision letter and accompanying author responses at the end of the HTML article, should your article be accepted for publication. In this event you acknowledge and agree that these will be published under the terms of the Creative Commons Attribution license.

Please use the following link to resubmit:

https://submit.elifesciences.org/cgi-bin/main.plex?el=A4Hz6Orc4A6HPh4I2A9ftdY6owynJMC0GdJVbv1MecnwZ

Best wishes,

Wendy Garrett
Senior Editor

On behalf of:
Eve Marder, Deputy Editor
Fiona Watt, Deputy Editor
Detlef Weigel, Deputy Editor
Randy Schekman, Editor-in-Chief

-----
Subject area information: SA List1 Major Subject Area(s): Developmental Biology and Stem Cells, Microbiology and Infectious Disease
SA List2 Research Organism(s): E. coli, Human


eLife Sciences Publications, Ltd is a limited liability non-profit non-stock corporation incorporated in the State of Delaware, USA, with company number 5030732, and is registered in the UK with company number FC030576 and branch number BR015634 at the address First Floor, 24 Hills Road, Cambridge CB2 1JP.
-----
* elisp script for custom TODO tracking in Org-mode
#+begin_src emacs-lisp :eval yes :results silent
;; execute this code block to add this agenda view to org-agenda menu
;; this only needs to be run once unless setq org-agenda-custom commands 
;; is altered elsewhere or this code block is updated
(add-to-list 'org-agenda-custom-commands
	'("r" "eLife revisions"
	  ((agenda ""
		   ((org-agenda-files 		     
		   '("/home/david/Data/HIO_Ecoli_paper/DOC/eLife/revisions/todo.org"))
		   (org-agenda-ndays 3)
		   (org-deadline-warning-days 14)
		    ))
	   (todo "TODO" ;alltodo ""
		 ((org-agenda-files 		   
		 '("/home/david/Data/HIO_Ecoli_paper/DOC/eLife/revisions/todo.org"))
		 (org-agenda-skip-function
		 '(org-agenda-skip-entry-if 'deadline 'scheduled))))
	   (todo "WAITING|IN-PROGRESS" ;alltodo ""
		 ((org-agenda-files 
		 '("/home/david/Data/HIO_Ecoli_paper/DOC/eLife/revisions/todo.org"))
		 (org-agenda-skip-function
		 '(org-agenda-skip-entry-if 'deadline 'scheduled))))
	   )
	  )
	)
#+end_src

* eLife revision due date [19%]
   DEADLINE: <2017-09-22 Fri -60d>
   :PROPERTIES:
   :COOKIE_DATA: todo recursive
   :END:
ESSENTIAL REVISIONS:
General comments on presentation in the paper
** TODO Revise methods reporting [0/6]
 Especially for a paper where the main point is development of an experimental system, the technical validation and reporting of the methods and results in the paper must be completely solid. The reviewers noted missing information on the number of biological and technical repeats, concentrations of stimuli, number of injected bacteria from the figure legends. The methods section was considered incomplete to allow a researcher from a related field to exactly reproduce every experiment shown - concentrations, solvents and timings - e.g. Fig 7, how much TNF, how much IFNgamma? It was unclear exactly how the false discovery rate was assessed - which algorithm was used? What sample numbers were used in the calculations and how reproducible were the experiments? Specific information about the E. coli strain should be given, and whether it is motile.

*** TODO Update number of biological and technical replicates in each experiment
" The reviewers noted missing information on the number of biological and technical repeats, concentrations of stimuli, number of injected bacteria from the figure legends."
*** TODO Update concentrations of stimulants and number of injected bacteria
 The reviewers noted missing information on the number of biological and technical repeats, concentrations of stimuli, number of injected bacteria from the figure legends.
*** TODO Describe FDR calculation
"It was unclear exactly how the false discovery rate was assessed - which algorithm was used? "
*** TODO Give specific information about E. coli strain
"Specific information about the E. coli strain should be given, and whether it is motile."
*** TODO Clarify the application of antibiotics
"The model depends on selective injection of live bacteria into HIOs and evidence is provided in Figure 1 about this. The experimental protocol needs to be far more explicit in the main body of the text: in the methods under 'HIO culture' it is stated 'HIOs were maintained in ENR media without antibiotics prior to microinjection experiments'; then under 'Microinjection', '..... cultures were rinsed with PBS and treated with ENR media containing penicillin and streptomycin...'. Controls that assess the potential influence of carry-over of antibiotics of the readouts reported in Figure 1 are required."
*** TODO Clarify dose for figure 1D
 " C. What dose(s) of injections were used for 1D?"

** TODO Generate control data that assess the influence of abx 
  Major comments
  The model depends on selective injection of live bacteria into HIOs and evidence is provided in Figure 1 about this. The experimental protocol needs to be far more explicit in the main body of the text: in the methods under 'HIO culture' it is stated 'HIOs were maintained in ENR media without antibiotics prior to microinjection experiments'; then under 'Microinjection', '..... cultures were rinsed with PBS and treated with ENR media containing penicillin and streptomycin...'. Controls that assess the potential influence of carry-over of antibiotics of the readouts reported in Figure 1 are required.
and ...
"Further, the final sentence of the third paragraph of page 4 and the results quoted in the following paragraph after day 3 are contradictory. Controls are absolutely necessary to ensure that these findings are not simply due to antibiotic carry-over."
** TODO Respond to fixation and/or freezing artifacts criticism of Fig 1B
"Fig 1B appears to be compromised by fixation and/or freezing artifacts."
** DONE Fix Fig 1C axis label
   SCHEDULED: <2017-08-01 Tue>
"B. In Figure 1C the y axis is labelled 24h delta CFU. Given that the scale becomes fractional, it seems likely that ratios rather than differences are being used."

RESPONSE: The reviewers are correct in pointing out that the y-axis label in Figure 1C was inaccurate. This has been amended in the revised manuscript. In addition, Figure 1 - Supplement 2 has been provided to clarify the exact relationship between the input CFU at t=0 and the CFU harvested from the lumen at 24 hours post microinjection

  Further issues on Figure 1.
  A. Fig 1B appears to be compromised by fixation and/or freezing artifacts.
  B. In Figure 1C the y axis is labelled 24h delta CFU. Given that the scale becomes fractional, it seems likely that ratios rather than differences are being used.
  C. What dose(s) of injections were used for 1D?
  D. As a main panel of Figure 1E, showing two agar plates does not add much, especially when the legend does not give details of exactly when are how they were generated. Further, the final sentence of the third paragraph of page 4 and the results quoted in the following paragraph after day 3 are contradictory. Controls are absolutely necessary to ensure that these findings are not simply due to antibiotic carry-over.
** TODO Consider removal or replacement of 1E
"As a main panel of Figure 1E, showing two agar plates does not add much, especially when the legend does not give details of exactly when are how they were generated."
** IN-PROGRESS [#A] Experiment evaluating HIO morphology and differentiation [4/8]

 "The transcriptomic approach does not capture all basic information about the system. Does the E. coli injection alter the size, morphology and longevity of the HIOs? Does it alter the rate of epithelial cell turnover (which could be easily tested by BrdU labeling, and would be predicted from e.g. Proc Natl Acad Sci U S A. 2011 Mar 15;108 Suppl 1:4570-7)? Does it alter the pattern of epithelial cell maturation, i.e. the fraction of mature paneth cells, goblet cells, enteroendocrine cells etc?"
*** IN-PROGRESS Treat and collect HIOs [66%]
   SCHEDULED: <2017-07-24 Mon>

**** DONE Spike HIOs with 10 uM EdU
     SCHEDULED: <2017-07-26 Wed 13:30>
     :PROPERTIES:
     :Effort:   0
     :END: 
   Entered on [2017-07-26 Wed 13:39]
**** DONE [#A] Collect HIOs and fix in PFA [24 hr timepoint]
     SCHEDULED: <2017-07-26 Wed 15:30>
     :PROPERTIES:
     :Effort:   0
     :END: 
**** DONE [#A] Collect HIOs and fix in PFA [48 hr timepoint]
     SCHEDULED: <2017-07-27 Thu 15:30>
     :PROPERTIES:
     :Effort:   0
     :END: 
**** DONE [#A] Collect HIOs and fix in PFA [96 hr timepoint]
     SCHEDULED: <2017-07-29 Sat 15:30>
     :PROPERTIES:
     :Effort:   0
     :END: 
**** TODO [#A] Collect HIOs and fix in PFA [192 hr timepoint]
     SCHEDULED: <2017-08-02 Wed 15:30>
     :PROPERTIES:
     :Effort:   0
     :END: 
   Entered on [2017-07-26 Wed 13:39]
**** TODO [#A] Submit HIOs for processing in MIL
     SCHEDULED: <2017-08-03 Thu>
*** TODO Give embedded HIOs to Mindy for sectioning
 The transcriptomic approach does not capture all basic information about the system. Does the E. coli injection alter the size, morphology and longevity of the HIOs? Does it alter the rate of epithelial cell turnover (which could be easily tested by BrdU labeling, and would be predicted from e.g. Proc Natl Acad Sci U S A. 2011 Mar 15;108 Suppl 1:4570-7)? Does it alter the pattern of epithelial cell maturation, i.e. the fraction of mature paneth cells, goblet cells, enteroendocrine cells etc?
*** TODO Does goblet cell differentiation contribute to mucus expression?
     In relation to figure 6, can induction of mucus production and induction of goblet cell differentiation be delineated? The slow appearance of mucin gene upregulation appears more consistent with a differentiation phenotype than simple gene expression?

** TODO Discuss "caveats" to studying hypoxia in absence of blood vessels
"It is accepted that enteroids are 3 dimensional structures, and the study has been productive in the evaluation of stem cell dynamics and other tissue level events. However, organoids are not vascularized with a capillary network that generate O2 gradients in vivo, thus organism level physiological events such as hypoxia are difficult to address and these caveats must be included in the discussion"
** TODO [#A] Detailed network analysis of GO and REACTOME pathways in Fig 2C
"Using the pathway analysis from the GO and REACTOME databases limits the insight that the reader is given into the changes in gene expression. For example, in Figure 2C 'muscle cell differentiation' is shown as a pathway. It would be possible to report detailed network analyses at least in the supporting material so that the contribution of different transcripts to the overall analysis is clear. This would allow considerable refinement of the time-dependent transcriptional response descriptions in the text."
** IN-PROGRESS [#A] Repeat PMDZ and E-Cad staining experiment [3/6]
   SCHEDULED: <2017-07-31 Mon>
  In Figure 3C it is difficult to appreciate whether the cadherin stain has worked in the PBS panel. How reproducible are these results? What happens with a hypoxic control?
*** DONE Request HIOs
    SCHEDULED: <2017-07-31 Mon>
*** DONE [#A] Setup HIOs for PMDZ uptake
    SCHEDULED: <2017-08-01 Tue>
    - PBS
    - E. coli
    - Hypoxia
    - dead E. coli
    - dead E. coli + hypoxia 
see also [[*Test%20BD-2%20induction%20in%20dead%20E.%20coli%20+%20hypoxia%20%5B/%5D][Test BD-2 induction in dead E. coli + hypoxia]]
*** DONE [#A] Start ECOR2 culture
    SCHEDULED: <2017-08-01 Tue>
*** TODO [#A] Microinject live/dead ECOR2 and start hypoxia treatment
    SCHEDULED: <2017-08-02 Wed>
*** TODO [#A] Collect media and fix HIOs in PFA
    SCHEDULED: <2017-08-03 Thu>
*** TODO [#A] Transfer HIOs to 70% Ethanol
    SCHEDULED: <2017-08-04 Fri>
** TODO Revise Figure 4 analysis [0/7]
   The data analysis shown in Fig 4. was complex and challenging for the reviewers to appreciate in its current form. The subset organisation must be clear and specific.
*** TODO [#A] Show PBS control +/- NF-kB inhibitor overlap with other NF-kB dependent subsets
    A. Why is there no PBS control with the NFkB inhibitor? Many cell survival/cell death pathways converge on NFkB signalling and the inhibitor may exert effects already at baseline which would then fall into gene set 1 and gene set 2? From the Venn-diagrams shown in panel A, it appears that genes in set 1 and 2 are mutually exclusive, which would make the lacking control less problematic, but apparently the hypoxia-responsive genes are highly enriched in both (of course these may be different genes in the same pathway? But this would be slightly surprising if a statistical exclusion had been made).
*** TODO Alter discussion to include "vital PAMPS"
    B. For the discussion of the analysis, it appears that an assumption is made that the effect of live E. coli is identical to dead E. coli combined with hypoxia. There is a large body of literature on "vital PAMPs" (e.g. Nature. 2011 May 22;474(7351):385-9), which might suggest there is more to it. Nevertheless, there appears to be an excellent correlation between genes regulated by live and dead E. coli in the plots in Fig. 4B. If the hypothesis is correct, then injecting dead E. coli and then immediately transferring the organoids to a hypoxic chamber should produce a gene-expression profile that correlates better that the dead E. coli alone. Noticeably the gradient of the correlation between hypoxia/PBS and live E. coli/PBS seems to be close to zero?
*** TODO Clarify criteria for scatterplots in 4B
    C. Is there some sort of statistical significance cut-off for the genes identified for each set in panel B? The clouds appear to very closely approach a log2-FC of zero, suggesting genes showing very small changes in expression are included? Would it be logical to show the data pre-filtered for p-value? Or show the limits of the region and color-code for p-value?
*** TODO Clarify labels for 4C
    D. In panel C, is it right that "% genes matched to pathway" is this the percentage of genes from each set (e.g. the full 1940 genes in Set 1) that map to the indicated pathway? Thus 5%, i.e. around 400 genes from set 1 map to "regulation of cytoskeleton organization"? Please clarify. 
*** TODO Supplementary figure showing size changes in gene expression in 4B
Also, as the plots in B suggest that many genes are included with a very small up- or down-regulation, it would be important to have some handle on not just the significance, but also the average absolute size of the change observed. A second set of graphs, or a supplementary figure with more information would be helpful.
*** TODO Clarify P-value cut-off for 4C
    SCHEDULED: <2017-08-02 Wed>
    E. In the legend, were pathways with enrichment P-values greater than 0.01 excluded?
** IN-PROGRESS [#A] Test BD-2 induction in dead E. coli + hypoxia [1/3]
   SCHEDULED: <2017-07-31 Mon>
    In Fig 5B and C, it would be important to include a group microinjected with dead E. coli, immediately followed by hypoxia, to conclude that both factors act together to induce b-defensins.
- cross-reference with [[*Repeat PMDZ and E-Cad staining experiment][Repeat PMDZ and E-Cad staining experiment {1/6}]]
*** DONE Request HIOs
*** TODO Submit Media to ELISA core
    SCHEDULED: <2017-08-03 Thu>
*** WAITING Analyze data
** TODO Respod to reducing agent question as beyond scope
For Fig 5D, does hBD-2 require a reducing agent for activity as hBD1 does (Nature 469, 419-423)? Typically, these pore-forming AMPs to exert a stronger effect on the rapidly growing bacteria than in the stationary phase, and in fact, your maximum growth rate (i.e. maximum curve gradient) is even higher where the BD-2 is added, suggesting that something in the BD-2 may even permit faster E coli growth in LB. Death over several hours in late stationary phase may be rather due to accumulation of a toxic metabolite. To control for these effects, it will be important to show growth data with heat-inactivation of the BD-2. 
** TODO [#A] Test recombinant BD-2 activity during stationary phase
To focus on killing in the stationary phase a late stationary-phase culture could be treated with differing concentrations of BD-2 over short time-courses (including the inactivated controls), measuring loss of membrane integrity by Sytox-green uptake by flow cytometry or microscopy. As O-antigens can inhibit AMP function, E. coli K-12 could be included in these experiments.
** TODO Prepare BD-2 supplementary figure with E. coli K12
** DONE Correct statement on epithelial barrier integrity
   SCHEDULED: <2017-08-02 Wed>
    The interpretation that 'Epithelial barrier integrity is enhanced following bacterial association' (p13) is rather at odds with the data in Figure 7 B and D where the PBS and E.coli treated permeability is the same. Is the meaning that NFkappaB signaling is required for the compensatory effects of the barrier in E.coli-treated organoids? 

RESPONSE: The reviewers are correct to note that E. coli colonization itself does not result in a decrease in barrier permeability compared to PBS, and that the presence of NF-kB inhibitor results in increased barrier permeability. This implies that NF-kB is required for the mediating the response to E. coli colonization. Our original intention was to highlight two findings: 1 - that maintenance of epithelial barrier integrity following E. coli colonization is NF-kB dependent, and 2 - that bacterial colonization mitigates damage to the epithelial barrier and increases in epithelial barrier permeability following exposure to pr-inflammatory cytokines. We orignially highlighted these two findings under a single header, stating that both were evidence of enhanced barrier integrity. The reviewer's comments, however, make it clear that these data are probably best presented as two distinct findings, and that is the approach we took in the 'Discussion'. We have therefore separated our description of these results into two distinct headers in the 'Results' section. We hope that this will clarify and highlight the effects of bacterial colonization on HIO barrier function and improve consistency with the interpretation we have laid out in the Discussion.
** TODO Prepare supplemental figure showing effect of SC514 on barrier function
What is the effect of the inhibitor alone?
** TODO Respond to minor points [0/6]
*** TODO Revise presentation of Figure 2C 
     The data shown in Fig 2C is potentially misleading. What you are most likely seeing here is a sigmoidal growth to a reasonably uniform carrying capacity of E. coli. The logarithmic growth rate is likely identical for all different inocula, but the plateau phase is reached earlier with the higher dose injections. Please simply plot the injected CFU and the final CFU per organoid. Ideally a time-course over the first 24h of colonization with a low and high dose of E. coli could be shown, giving you a more detailed handle on the type of growth occurring within the organoids.

RESPONSE: Figure 1 - Supplement 2 has been provided to clarify the exact relationship between the input CFU at t=0 and the CFU harvested from the lumen at 24 hours post microinjection

*** TODO Provide explanation for BD1 expression
     What is the explanation for the increase in BD1 over time in the PBS injected organoids (Figure 2E which lacks units)? It would be interesting to include control panels here for cell death (LDH release or similar) and total cell number per sample volume.
*** TODO Address function of fucosylated mucins
     Fucosylation of intestinal glycans has been reported to depend upon the presence of bacterial pathobionts or pathogens (e.g. SFB, Salmonella) and the activity of innate lymphoid cells in the lamina propria (e.g. Science 12 Sep 2014:Vol. 345, Issue 6202, 1254009), or to be associated with mature M cells. "Mature carbohydrate modifications" may be something of an oversimplification, and this observation may even suggest that the closed nature of the organoids leads to a non-pathogenic E. coli strain being recognized as quite aggressive.
*** TODO Clarify principle component analysis
    SCHEDULED: <2017-08-02 Wed>
     Page 3, second paragraph. The description of the principal component analysis should read that PC1 correlates with developmental stage, while PC2 correlates with tissue maturation status. Currently the text implies that there is a direct relationship.
*** TODO Include suggested references
     The authors should be accurate in acknowledging older literature (the hypoxia NFkappaB effect has been known since 1994 and the BD-2 effect since 1997) - albeit not in this context.

** IN-PROGRESS [#A] Authenticate cell lines [1/3]
   SCHEDULED: <2017-08-01 Tue>
    If your work involved the use of cell lines, please indicate in the Materials and methods section of your manuscript if their identity has been authenticated, state the authentication method (such as STR profiling), and report the mycoplasma contamination testing status. Please consult our Journal Policies for further details: https://submit.elifesciences.org/html/elife_author_instructions.html#policies

*** DONE Start ES cell culture 
    SCHEDULED: <2017-08-01 Tue>

*** TODO Harvest ES cells and submit to core
    SCHEDULED: <2017-08-04 Fri>
 Protocol here:
 [[https://umich.app.box.com/file/204465078839]]

** WAITING Update ArrayExpress publication details
[[mu4e:msgid:1178471856.15.1501632037672.JavaMail.arrayexpress@ebi.ac.uk][E-MTAB-5801, ArrayExpress release in 30 days time]]
